src/lsp/protocol/response_error.cr

```diff
@@ -17,15 +15,33 @@ module LSP::Protocol
     RequestCancelled = -32800
   end
 
+  class ProtocolException < Exception
+    getter code
+
+    def initialize(
+      @message : String? = nil,
+      @cause : Exception? = nil,
+      @code : ErrorCodes = ErrorCodes::UnknownErrorCode
+    )
```

Comment:
Instead of setting the ivars you don't own (message & cause) directly, I would suggest using `super` to set them, and only handle your added ivar (code).

Something like:
```cr
def initialize(message, cause, @code : ErrorCodes = ErrorCodes::UnknownErrorCode)
  super(message, cause)
end
```

This way this exception class will still work if `Exception` changes in the future (like by adding an ivar, which would require a change here with your current code).